lab-kyle/test/note-route-test.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const request = require('superagent');
+const fs = require('fs');
+const expect = require('chai').expect;
+const server = require('../server');
+const port = process.env.PORT || 3000;
+const serverUrl = `http://localhost:${port}`;
+
+
+describe('testing note-route module', function(){
+  before(function(done) {
+    if(!server.isRunning){
+      server.listen(port, function(){
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after(function(done){
+    if (server.isRunning){
+      server.close(function(){
+        server.isRunning = false;
+        console.log('server shut down');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+//post testing
+  describe('testing method POST on endpoint /api/note', function(){
+    before((done)=>{
+      console.log('serverUrl', serverUrl);
+      request.post(`${serverUrl}/api/note`)
+        .send({content: 'hello'})
+        .end((err, res) => {
+          this.res = res;
+          this.note = res.body;
+          done();
+        });
+    });
+
+    it('should return status 200', ()=>{
+      expect(this.res.status).to.equal(200);
+    });
+
+    it('should return a note', ()=>{
+      expect(this.note.content.content).to.equal('hello');
```

Comment:
This is okay, but kind of silly :D, see my comment on your note-route.js for more info on how you may go about avoiding this sort of ambiguity in the future.